sql/core/src/main/java/org/apache/spark/sql/sources/v2/writer/DataWriter.java

```diff
@@ -84,9 +86,9 @@
    * This method will only be called if there is one record failed to write, or {@link #commit()}
    * failed.
    *
-   * If this method fails(throw exception), the underlying data source may have garbage that need
-   * to be cleaned by {@link DataSourceV2Writer#abort(WriterCommitMessage[])} or manually, but
-   * these garbage should not be visible to data source readers.
+   * If an exception was thrown, the underlying data source may have garbage that need to be
+   * cleaned by {@link DataSourceV2Writer#abort(WriterCommitMessage[])} or manually, but these
+   * garbage should not be visible to data source readers.
    */
   void abort();
```

Comment:
May as well declare throws IOE here too, because things invoked are likely to declare that, and whatever calls this is going to need to be catching and downgrading all exceptions anyway